openfl/utilities/attestation/attestation_utils.py

```diff
@@ -0,0 +1,259 @@
+        ecdsa_p_384_signer = ECDSASigner.get_instance()
+        cert = ecdsa_p_384_signer.cert("localhost",mrenclave).encode('utf-8')
+        key = ecdsa_p_384_signer.serialize_private_key()
+        pub_key = ecdsa_p_384_signer.get_pubkey()
+        pubkey_path = os.path.join(self.attestation_report_path, f"{self.participant_name}_pubkey.pem")
+        with open(pubkey_path, "wb") as fh:
+            fh.write(pub_key.encode('utf-8'))
+        avs_report = self.get_avs_report_ita(
+            self.participant_name,
+            self.attestation_report_path,
+            cert,
+            self.avs_url,
+            self.ita_api_key
+        )
+        logger.info(f"AVS report: {avs_report}")
+        return {'mrenclave': mrenclave,
+                'name': self.participant_name,
+                'token': avs_report["token"]}
+        
+
+    def gen_sgx_quote(
+                self,
+                user_data_bytes=None,
+                quote_dump_path='/tmp/quote.json',
+                orig_data=None,
+                challenge="0000000000000"):
+        """Generates the SGX quote and saves it to the specified path.
+        Args:
+            user_data_bytes (bytes): User data to be included in the quote.
+            quote_dump_path (str): Path to save the generated quote.
+            orig_data (bytes): Original data to be included in the quote.
+            challenge (str): Challenge string to be included in the quote.
+        Returns:
+            dict: A dictionary containing the generated quote.
+        Raises:
+            ValueError: If the user data is not a bytes object or exceeds 64 bytes.
+        """
+
+        # Set user data
+        if user_data_bytes is not None:
+            # Check that user data is a bytes like object
+            if isinstance(user_data_bytes, bytes) is not True:
+                raise ValueError("User data must be a bytes object")
+
+            # Check that user data is at most 64 bytes
+            if len(list(user_data_bytes)) > 64:
+                raise ValueError("User data can be at most 64 bytes")
+
+            # Set user report data
+            with open("/dev/attestation/user_report_data", 'wb') as fh:
+                fh.write(user_data_bytes)
+
+        # Generate attestation quote
+        quote = None
+        with open("/dev/attestation/quote", 'rb') as fh:
+            quote = fh.read(8192)
+
+        # Create quote as expected by AVS for verification
+        # a. base64 encode the quote
+        # b. Set 'userData' to empty, we do not want AVS to verify user data
+        quote_avs = {}
+        quote_avs["quote"] = base64.b64encode(quote).decode('utf-8')
+
+        if orig_data is not None:
+            quote_avs['runtime_data'] = base64.b64encode(orig_data).decode('utf-8')
+        else:
+            quote_avs['runtime_data'] = ''
+
+        # Save the quote in the specified location
+        with open(quote_dump_path, 'w') as fh:
+            json.dump(quote_avs, fh)
```

Comment:
Just wondering, why do we need to persist the quote in a file? AFAIK, it's only needed when asking ITA for remote attestation.

Reply:
As part of the OpenFL-Security refactoring, I suggest trying to remove this and see if it affects the overall flow.